### PR TITLE
Implement UI for Computation Resource Properties

### PR DESCRIPTION
**Pull request by c-mueller**
This PR implements a list view for ComputationResource Properties, a creation dialog and a confirm dialog.

The list is a generic component that can be reused if necessary. Currently the lists are implemented for Implementations and Algorithms.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,8 +7,10 @@
     "build": "ng build",
     "test": "ng test",
     "lint": "eslint ./src --ext .js,.ts",
+    "lint-fix": "eslint ./src --fix --ext .js,.ts",
     "gen": "ng-openapi-gen --input http://localhost:8080/atlas/v3/api-docs -o generated/api",
     "postgen": "prettier generated/api --write",
+    "gen-api": "ng-openapi-gen --input http://localhost:8080/atlas/v3/api-docs -o generated/api && prettier generated/api --write",
```
Comment (timniederhausen) [inline package.json:13]: Why did you add this?
gen & postgen should already handle this.

File: generated/api/api.module.ts [hunk 2]
```diff
@@ -8,8 +8,8 @@ import {
 import { HttpClient } from '@angular/common/http';
 import { ApiConfiguration, ApiConfigurationParams } from './api-configuration';
 
-import { AlgorithmService } from './services/algorithm.service';
 import { AlgorithmRelationTypeService } from './services/algorithm-relation-type.service';
+import { AlgorithmService } from './services/algorithm.service';
```
Comment (timniederhausen) [inline generated/api/api.module.ts:12]: hmm, did you use the current ng-openapi-fork?
All of that should be sorted by now.

File: package.json (hunk 1 above)
Comment (FlxB2) [inline package.json:13]: This shouldn't be necessary gen + postgen does the same

File: src/app/components/generics/tree-output/tree-output.component.scss [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+.tree-wrapper{
+  margin: 16px;
```
Comment (FlxB2) [inline src/app/components/generics/tree-output/tree-output.component.scss:3]: Boostrap margins should be used if possible

File: package.json (hunk 1 above)
Comment (c-mueller) [reply package.json:13]: Laziness and to ensure both commands have been run.

File: generated/api/api.module.ts (hunk 2 above)
Comment (c-mueller) [reply generated/api/api.module.ts]: I did rerun it after npm install, maybe it looks good now.

File: package.json (hunk 1 above)
Comment (c-mueller) [reply package.json:13]: Well typing one command is faster than two. This is just to make sure both commandy have been run and I dont foget one.

File: src/app/components/generics/tree-output/tree-output.component.scss (hunk 3 above)
Comment (c-mueller) [reply src/app/components/generics/tree-output/tree-output.component.scss:3]: I have no idea why this file has been changed. I certainly, at least not by intention, changed anything in it.

File: package.json (hunk 1 above)
Comment (FlxB2) [reply package.json:13]: after running `gen`, `postgen` is executed automatically anyway as far as I know

Comment (c-mueller) [reply package.json]: > after running `gen`, `postgen` is executed automatically anyway as far as I know

Looks like it. Did remove the command then.
